gufe/protocols/protocoldag.py

```diff
@@ -43,18 +54,38 @@ def name(self):
 
     @property
     def graph(self) -> nx.DiGraph:
-        """DAG of `ProtocolUnit`s that produced this `ProtocolDAGResult`.
+        """DAG of `ProtocolUnit`s that comprise this object.
 
         """
         return self._graph
 
     @property
-    def protocol_units(self):
+    def protocol_units(self) -> list[ProtocolUnit]:
         """List of `ProtocolUnit`s given in DAG-order.
 
         """
         return list(self._iterate_dag_order(self._graph))
 
+    @property
+    def transformation_key(self) -> Union[GufeKey, None]:
+        """The `GufeKey` of the `Transformation` this object performs.
+
+        This functions as a label, indicating where this object came from.
+
+        """
+        return self._transformation_key
+
+    @property
+    def extends_key(self) -> Union[GufeKey, None]:
+        """The `GufeKey` of the `ProtocolDAGResult` this object extends.
+
+        This functions as a label, indicating where this object came from.
+        It can be used to reconstruct the set of extension relationships
+        between a collection of ProtocolDAG
+
```

Comment:
also needs documenting that this (and above) are optional.  Extends being `None` indicating it's not an extension, `transformation_key` being None indicating ????

Reply:
Done! Thanks for this!